causalpy/pymc_experiments.py

```diff
@@ -883,3 +884,129 @@ def _get_treatment_effect_coeff(self) -> str:
+        self._x_design_info = X.design_info
+        self.labels = X.design_info.column_names
+        self.y, self.X = np.asarray(y), np.asarray(X)
+        self.outcome_variable_name = y.design_info.column_names[0]
+
+        t, Z = dmatrices(instruments_formula, self.instruments_data)
+        self._t_design_info = t.design_info
+        self._z_design_info = Z.design_info
+        self.labels_instruments = Z.design_info.column_names
+        self.t, self.Z = np.asarray(t), np.asarray(Z)
+        self.instrument_variable_name = t.design_info.column_names[0]
+
+        self.get_naive_OLS_fit()
+        self.get_2SLS_fit()
+
+        # fit the model to the data
+        COORDS = {"instruments": self.labels_instruments, "covariates": self.labels}
+        self.coords = COORDS
+        if priors is None:
+            priors = {
+                "mus": [self.ols_beta_first_params, self.ols_beta_second_params],
+                "sigmas": [1, 1],
+                "eta": 2,
+                "lkj_sd": 2,
+            }
+        self.priors = priors
+        self.model.fit(
+            X=self.X, Z=self.Z, y=self.y, t=self.t, coords=COORDS, priors=self.priors
+        )
+
+    def get_2SLS_fit(self):
+        first_stage_reg = sk_lin_reg().fit(self.Z, self.t)
+        fitted_Z_values = first_stage_reg.predict(self.Z)
+        X2 = self.data.copy(deep=True)
+        X2[self.instrument_variable_name] = fitted_Z_values
+        _, X2 = dmatrices(self.formula, X2)
+        second_stage_reg = sk_lin_reg().fit(X=X2, y=self.y)
+        betas_first = list(first_stage_reg.coef_[0][1:])
+        betas_first.insert(0, first_stage_reg.intercept_[0])
+        betas_second = list(second_stage_reg.coef_[0][1:])
+        betas_second.insert(0, second_stage_reg.intercept_[0])
+        self.ols_beta_first_params = betas_first
+        self.ols_beta_second_params = betas_second
+        self.first_stage_reg = first_stage_reg
+        self.second_stage_reg = second_stage_reg
+
+    def get_naive_OLS_fit(self):
+        ols_reg = sk_lin_reg().fit(self.X, self.y)
+        beta_params = list(ols_reg.coef_[0][1:])
+        beta_params.insert(0, ols_reg.intercept_[0])
+        self.ols_beta_params = dict(zip(self._x_design_info.column_names, beta_params))
+        self.ols_reg = ols_reg
+
+    def _input_validation(self):
+        """Validate the input data and model formula for correctness"""
+        treatment = self.instruments_formula.split("~")[0]
+        test = treatment.strip() in self.instruments_data.columns
+        test = test & (treatment.strip() in self.data.columns)
+        if not test:
+            raise DataException(
+                f"""
+                The treatment variable:
+                {treatment} must appear in the instrument_data to be used
+                as an outcome variable and in the data object to be used as a covariate.
+                """
+            )
+        check_binary = len(self.data[treatment.strip()].unique()) > 2
```

Comment:
This check is actually working fine. It just so happens that the treatment variable `risk` in the model is a score between 0-10. The treatment effect here should be thought of as a dosage treatment rather than a binary treatment here. I've added a line to the write up mentioning this.

Reply:
<img width="745" alt="image" src="https://github.com/pymc-labs/CausalPy/assets/10588167/fcedd203-dcc3-4215-8160-fb567c68bf16">

Reply:
Ah ok. So I misunderstood before - I didn't read that closely and assumed it _had_ to be binary. If that's not the case, then I guess we can remove that exception? Sorry about that.

Reply:
Let's leave it in. I think many of the use-cases will be binary treatments, and it's no harm to be reminded to take care interpreting coefficients. Can revisit when i implement the sklearn/statsmodels IV class  if you like.